examples/fluids/problems/sgs_dd_model.c

```diff
@@ -150,8 +164,259 @@ PetscErrorCode SgsDDModelSetupNodalEvaluation(Ceed ceed, User user, CeedData cee
+  PetscFunctionReturn(PETSC_SUCCESS);
+}
+
+// @brief Create CeedOperator to calculate data-drive SGS at nodes using sequentially-applied operators
+static PetscErrorCode SgsDDSetupNodalEvaluation_Sequential(Ceed ceed, User user, CeedData ceed_data, SgsDDSetupData sgs_dd_setup_data) {
+  SgsDDData           sgs_dd_data = user->sgs_dd_data;
+  CeedInt             num_comp_q, num_comp_grad_velo, num_comp_x, num_comp_grid_aniso, num_comp_eigvec = 9 + 1;
+  PetscInt            dim;
+  CeedVector          inv_multiplicity, eigvec;
+  CeedElemRestriction elem_restr_inv_multiplicity, elem_restr_grad_velo, elem_restr_sgs, elem_restr_eigvec, elem_restr_dd_inputs,
+      elem_restr_dd_outputs;
+  DMLabel  domain_label = NULL;
+  PetscInt label_value = 0, height = 0, dm_field = 0;
+
+  PetscFunctionBeginUser;
+  {  // Create DMs for data-driven input and output values
+    PetscSection section;
+    PetscInt     degree, q_extra;
+    {  // Get degree and number of quadrature points from dm_sgs
+      PetscFE         fe;
+      PetscSpace      basis;
+      PetscQuadrature quadrature;
+      PetscInt        num_qpnts;
+      PetscCall(DMGetField(sgs_dd_data->dm_sgs, 0, NULL, (PetscObject *)&fe));
+      PetscCall(PetscFEGetBasisSpace(fe, &basis));
+      PetscCall(PetscSpaceGetDegree(basis, &degree, NULL));
+      PetscCall(PetscFEGetQuadrature(fe, &quadrature));
+      PetscCall(PetscQuadratureGetOrder(quadrature, &num_qpnts));
+      q_extra = degree - num_qpnts;
+    }
+
+    PetscCall(DMClone(sgs_dd_data->dm_sgs, &sgs_dd_data->dm_dd_inputs));
+    PetscCall(PetscObjectSetName((PetscObject)sgs_dd_data->dm_dd_inputs, "Data-Driven Model Inputs"));
+    PetscCall(DMSetupByOrder_FEM(PETSC_TRUE, PETSC_TRUE, degree, 1, q_extra, 1, &sgs_dd_data->num_comp_inputs, sgs_dd_data->dm_dd_inputs));
+    PetscCall(DMGetLocalSection(sgs_dd_data->dm_dd_inputs, &section));
+    PetscCall(PetscSectionSetFieldName(section, 0, ""));
+    for (CeedInt i = 0; i < sgs_dd_data->num_comp_inputs; i++) {
+      char component_name[PETSC_MAX_PATH_LEN];
+      PetscCall(PetscSNPrintf(component_name, sizeof component_name, "DataDrivenInput%" CeedInt_FMT, i + 1));
+      PetscCall(PetscSectionSetComponentName(section, 0, i, component_name));
+    }
+
+    PetscCall(DMClone(sgs_dd_data->dm_sgs, &sgs_dd_data->dm_dd_outputs));
+    PetscCall(PetscObjectSetName((PetscObject)sgs_dd_data->dm_dd_outputs, "Data-Driven Model Outputs"));
+    PetscCall(DMSetupByOrder_FEM(PETSC_TRUE, PETSC_TRUE, degree, 1, q_extra, 1, &sgs_dd_data->num_comp_outputs, sgs_dd_data->dm_dd_outputs));
+    PetscCall(DMGetLocalSection(sgs_dd_data->dm_dd_outputs, &section));
+    PetscCall(PetscSectionSetFieldName(section, 0, ""));
+    for (CeedInt i = 0; i < sgs_dd_data->num_comp_outputs; i++) {
+      char component_name[PETSC_MAX_PATH_LEN];
+      PetscCall(PetscSNPrintf(component_name, sizeof component_name, "DataDrivenOutput%" CeedInt_FMT, i + 1));
```

Comment:
```suggestion
    for (CeedInt i = 0; i < sgs_dd_data->num_comp_inputs; i++) {
      char component_name[PETSC_MAX_PATH_LEN];
      
      PetscCall(PetscSNPrintf(component_name, sizeof component_name, "DataDrivenInput%" CeedInt_FMT, i + 1));
      PetscCall(PetscSectionSetComponentName(section, 0, i, component_name));
    }

    PetscCall(DMClone(sgs_dd_data->dm_sgs, &sgs_dd_data->dm_dd_outputs));
    PetscCall(PetscObjectSetName((PetscObject)sgs_dd_data->dm_dd_outputs, "Data-Driven Model Outputs"));
    PetscCall(DMSetupByOrder_FEM(PETSC_TRUE, PETSC_TRUE, degree, 1, q_extra, 1, &sgs_dd_data->num_comp_outputs, sgs_dd_data->dm_dd_outputs));
    PetscCall(DMGetLocalSection(sgs_dd_data->dm_dd_outputs, &section));
    PetscCall(PetscSectionSetFieldName(section, 0, ""));
    for (CeedInt i = 0; i < sgs_dd_data->num_comp_outputs; i++) {
      char component_name[PETSC_MAX_PATH_LEN];
      
      PetscCall(PetscSNPrintf(component_name, sizeof component_name, "DataDrivenOutput%" CeedInt_FMT, i + 1));
```
minor - usually we have newlines after variable declaration blocks